installer/scripts/tailfilereader.rb

```diff
@@ -0,0 +1,396 @@
+require 'optparse'
+require 'logger'
+
+module Tailscript
+
+  class NewTail  
+    def initialize(paths)
+      @paths = paths
+      @tails = {}
+      @pos_file = $options[:pos_file] 
+      @read_from_head = $options[:read_from_head]
+      @pf = nil
+      @pf_file = nil
+
+      $log = Logger.new(STDERR)
+      $log.formatter = proc do |severity, time, progname, msg|
+        "#{severity} #{msg}\n"
+      end
+    end
+
+    attr_reader :paths
+
+    def start
+      target_paths = expand_paths
+      start_watchers(target_paths) unless target_paths.empty?
+    end
+
+    def shutdown
+      @pf_file.close if @pf_file
+    end
+
+    def expand_paths
+      date = Time.now      
+      paths = []
+
+      @paths.each { |path|
+        path = date.strftime(path)
+        if path.include?('*')
+          paths += Dir.glob(path).select { |p|
+            if File.readable?(p)
+              true
+            else
+              $log.warn "#{p} is unreadable. It is excluded and would be examined next time."
+              false
+            end
+          }
+        elsif !File.readable?(path)
+          $log.error "#{path} is unreadable. Cannot tail the file."
+        else
+          # When file is not created yet, Dir.glob returns an empty array. So just add when path is static.
+          paths << path
+        end
+      }
+    end
+
+    def setup_watcher(path, pe)
+      tw = TailWatcher.new(path, pe, @read_from_head, &method(:receive_lines))
+      tw.on_notify
+      tw
+    end
+
+    def start_watchers(paths)
+      if @pos_file
+        @pf_file = File.open(@pos_file, File::RDWR|File::CREAT)
+        @pf_file.sync = true
+        @pf = PositionFile.parse(@pf_file)
+      end
+
+      paths.each { |path|
+        pe = nil
+        if @pf
+          pe = @pf[path]    #pe is FilePositionEntry instance
+          if @read_from_head && pe.read_inode.zero?
+            begin
+              pe.update(File::Stat.new(path).ino, 0)
+            rescue Errno::ENOENT
+              $log.warn "#{path} not found. Continuing without tailing it."
+            end
+          end
+        end
+        
+        @tails[path] = setup_watcher(path, pe)
+      }
+    end
+
+    def receive_lines(lines, tail_watcher)
```

Comment:
tail_watcher is unused. I don't see the use for this function except print debugging